src/compiler/scala/tools/nsc/backend/jvm/BCodeHelpers.scala

```diff
@@ -1034,21 +1034,34 @@ object BCodeHelpers {
   }
 
   /**
-   * Contains helpers around converting a Scala signature (array of bytes) into an array of `Long`.
-   *  Details about the storage format of pickles at the bytecode level (classfile annotations) can be found in SIP-10.
+   * Helpers for encoding a Scala signature (array of bytes) into a String or, if too large, an
+   * array of Strings.
+   *
+   * The encoding is as described in [[scala.reflect.internal.pickling.ByteCodecs]]. However, the
+   * special encoding of 0x00 as 0xC0 0x80 is not done here, as the resulting String(s) are passed
+   * as annotation argument to ASM, which will perform this step.
    */
-  class ScalaSigBytes(bytes: Array[Byte]) {
+  final class ScalaSigBytes(bytes: Array[Byte]) {
```

Comment:
We could generalize this to take `bytes` and `len` to avoid an intermediate array creation in:

https://github.com/scala/scala/blob/2f3791c3079d998d29788d121552c27517f58a6c/src/compiler/scala/tools/nsc/backend/jvm/BCodeHelpers.scala#L405